%% Generated by Sphinx.
\def\sphinxdocclass{report}
\documentclass[letterpaper,10pt,english]{sphinxmanual}
\ifdefined\pdfpxdimen
   \let\sphinxpxdimen\pdfpxdimen\else\newdimen\sphinxpxdimen
\fi \sphinxpxdimen=.75bp\relax
\ifdefined\pdfimageresolution
    \pdfimageresolution= \numexpr \dimexpr1in\relax/\sphinxpxdimen\relax
\fi
%% let collapsible pdf bookmarks panel have high depth per default
\PassOptionsToPackage{bookmarksdepth=5}{hyperref}

\PassOptionsToPackage{booktabs}{sphinx}
\PassOptionsToPackage{colorrows}{sphinx}

\PassOptionsToPackage{warn}{textcomp}
\usepackage[utf8]{inputenc}
\ifdefined\DeclareUnicodeCharacter
% support both utf8 and utf8x syntaxes
  \ifdefined\DeclareUnicodeCharacterAsOptional
    \def\sphinxDUC#1{\DeclareUnicodeCharacter{"#1}}
  \else
    \let\sphinxDUC\DeclareUnicodeCharacter
  \fi
  \sphinxDUC{00A0}{\nobreakspace}
  \sphinxDUC{2500}{\sphinxunichar{2500}}
  \sphinxDUC{2502}{\sphinxunichar{2502}}
  \sphinxDUC{2514}{\sphinxunichar{2514}}
  \sphinxDUC{251C}{\sphinxunichar{251C}}
  \sphinxDUC{2572}{\textbackslash}
\fi
\usepackage{cmap}
\usepackage[T1]{fontenc}
\usepackage{amsmath,amssymb,amstext}
\usepackage{babel}



\usepackage{tgtermes}
\usepackage{tgheros}
\renewcommand{\ttdefault}{txtt}



\usepackage[Bjarne]{fncychap}
\usepackage{sphinx}

\fvset{fontsize=auto}
\usepackage{geometry}


% Include hyperref last.
\usepackage{hyperref}
% Fix anchor placement for figures with captions.
\usepackage{hypcap}% it must be loaded after hyperref.
% Set up styles of URL: it should be placed after hyperref.
\urlstyle{same}

\addto\captionsenglish{\renewcommand{\contentsname}{Contents:}}

\usepackage{sphinxmessages}
\setcounter{tocdepth}{1}



\title{MAFtoolbox}
\date{Jul 11, 2023}
\release{0.5}
\author{Christopher Klapproth}
\newcommand{\sphinxlogo}{\vbox{}}
\renewcommand{\releasename}{Release}
\makeindex
\begin{document}

\ifdefined\shorthandoff
  \ifnum\catcode`\=\string=\active\shorthandoff{=}\fi
  \ifnum\catcode`\"=\active\shorthandoff{"}\fi
\fi

\pagestyle{empty}
\sphinxmaketitle
\pagestyle{plain}
\sphinxtableofcontents
\pagestyle{normal}
\phantomsection\label{\detokenize{index::doc}}


\sphinxAtStartPar
\sphinxstylestrong{MAFtoolbox} is a Software written in Python that implements
a range of operations and transformations on genome alignments in the Multiple Alignment
Format (MAF). Examples of use case include the extraction of alignment subblocks
based on gene annotations, filtering of sequences based on identity and
merging of fragmented neighboring alignment blocks into lnger, coherent blocks.

\begin{sphinxadmonition}{note}{Note:}
\sphinxAtStartPar
This project is currently under development.
\end{sphinxadmonition}


\chapter{Indices and tables}
\label{\detokenize{index:indices-and-tables}}\begin{itemize}
\item {} 
\sphinxAtStartPar
\DUrole{xref,std,std-ref}{genindex}

\item {} 
\sphinxAtStartPar
\DUrole{xref,std,std-ref}{modindex}

\item {} 
\sphinxAtStartPar
\DUrole{xref,std,std-ref}{search}

\end{itemize}


\chapter{Installation}
\label{\detokenize{index:installation}}
\sphinxAtStartPar
While a bioconda installation is planned for future release, right now it
is heavily suggested to use the precompiled executable built with pyinstaller.
Download and unpack the archive, then navigate to the distributed binaries folder with:

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{g+go}{cd MAFtoolbox/dist/MAFtools}
\end{sphinxVerbatim}

\sphinxAtStartPar
You can test the executable with:

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{g+go}{./MAFtools \PYGZhy{}\PYGZhy{}help}
\end{sphinxVerbatim}

\sphinxAtStartPar
This should produce a short list of programs executable with MAFtools. To get more
information on one (here, as an example, for extracting alignment blocks according to
genome coordinates), you can type:

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{g+go}{./MAFtools extract \PYGZhy{}\PYGZhy{}help}
\end{sphinxVerbatim}

\sphinxAtStartPar
As it is obviously annoying to only use MAFtools from inside the download directory,
I would suggest setting up an alias for now, like this:

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{g+go}{alias MAFtools=\PYGZdl{}(pwd)/MAFtools}
\end{sphinxVerbatim}

\sphinxAtStartPar
As mentioned, a full installer automating this process will follow.


\chapter{Example usage}
\label{\detokenize{index:example-usage}}
\sphinxAtStartPar
The MAFtoolbox comes with a few example files, that can be used to play around
and get to know basic functionality. One useful application might be
highlighting the part of an alignment that includes some annotated gene.
The program

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{g+go}{MAFtools highlight}
\end{sphinxVerbatim}

\sphinxAtStartPar
exists for this purpose. In the most simple case, we can simply provide a MAF
alignment file and a corresponding annotation file in bed format. Note that
the sequence names (for example chrX, chrYâ€¦) in the annotation file need to exactly correspond with
the sequence names found in the alignment file to be found.
As a simple showcase, move to the root directory of the MAFtoolbox archive and type

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{g+go}{MAFtools highlight \PYGZhy{}\PYGZhy{}maf Examples/Apoidea\PYGZus{}genome\PYGZus{}tRNA\PYGZus{}blocks\PYGZus{}filtered.maf \PYGZhy{}\PYGZhy{}bed Examples/Amel\PYGZus{}tRNA\PYGZus{}examples.bed}
\end{sphinxVerbatim}

\sphinxAtStartPar
The output should display the alignments in MAF format, but with the coordinates corresponding to
the genes found in the .bed file highlighted in green. What if we want to highlight the
annotated genes \sphinxhyphen{}and\sphinxhyphen{} an additional 5 nucleotides (with respect to the reference sequence)
in both directions? We can use the \sphinxhyphen{}s (\textendash{}sense) and \sphinxhyphen{}n (\textendash{}antisense) parameters to add any
number of nucleotides to be highlighted:

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{g+go}{MAFtools highlight \PYGZhy{}\PYGZhy{}maf Examples/Apoidea\PYGZus{}genome\PYGZus{}tRNA\PYGZus{}blocks\PYGZus{}filtered.maf \PYGZhy{}\PYGZhy{}bed Examples/Amel\PYGZus{}tRNA\PYGZus{}examples.bed \PYGZhy{}s 5 \PYGZhy{}n 5}
\end{sphinxVerbatim}

\sphinxAtStartPar
You will notice that the highlighted regions are now enlarged corresponding to the \sphinxhyphen{}s and \sphinxhyphen{}n
parameters, but these overhang regions will be colored the same way as the annotated sequence.
To allow for visual distinction we can give the overhang regions another color, for example red:

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{g+go}{MAFtools highlight \PYGZhy{}\PYGZhy{}maf Examples/Apoidea\PYGZus{}genome\PYGZus{}tRNA\PYGZus{}blocks\PYGZus{}filtered.maf \PYGZhy{}\PYGZhy{}bed Examples/Amel\PYGZus{}tRNA\PYGZus{}examples.bed \PYGZhy{}s \PYGZhy{}n 5 \PYGZhy{}\PYGZhy{}overhang\PYGZhy{}color RED}
\end{sphinxVerbatim}

\sphinxAtStartPar
Now it should be readily visible what is what.

\sphinxAtStartPar
You can always explore all program options and parameters with the \textendash{}help function.



\renewcommand{\indexname}{Index}
\printindex
\end{document}